assets/js/util/urls.js

```diff
@@ -85,3 +85,27 @@ export function normalizeURL( incomingURL ) {
 export function isHashOnly( url ) {
 	return /^#\w[A-Za-z0-9-_]*$/.test( url );
 }
+
+/**
+ * Shortens a URL to fit a given length.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} url      The original URL to shorten.
+ * @param {number} maxChars The maximum length of the URL.
+ * @return {string} The shortened URL.
+ */
+export function shortenURL( url, maxChars ) {
+	if ( url.length <= maxChars ) {
+		return url;
+	}
+
+	const urlObject = new URL( url );
```

Comment:
`new URL` will throw if an invalid URL is provided to it which could happen if a selector that provides a valid URL hasn't resolved yet for example.

Let's wrap this in a try/catch or add a guard using `isURL` from `@wordpress/url` (essentially the same thing in a utility).